cognite_toolkit/_cdf_tk/utils/fileio/_readers.py

```diff
@@ -49,8 +52,32 @@ def from_filepath(cls, filepath: Path) -> "FileReader":
         )
 
 
+class NDJsonReader(FileReader):
+    format = ".ndjson"
+
+    def _read_chunks_from_file(self, file: TextIOWrapper) -> Iterator[JsonVal]:
+        for line in file:
+            if stripped := line.strip():
+                yield json.loads(stripped)
+
+
+class YAMLBaseReader(FileReader, ABC):
+    def _read_chunks_from_file(self, file: TextIOWrapper) -> Iterator[JsonVal]:
+        yield from yaml.safe_load_all(file)
+
+
+class YAMLReader(YAMLBaseReader):
+    format = ".yaml"
+
+
+class YMLReader(YAMLBaseReader):
+    format = ".yml"
```

Comment:
Nitpick (a bit):
Any reason why split the YAMLReader in two different classes? I see that things get really flexible to extend, but don't see the need in this specific case.

```
class YAMLReader(FileReader):
    # This class now handles both common YAML extensions
    formats = (".yaml", ".yml")

    def _read_chunks_from_file(self, file: TextIOWrapper) -> Iterator[JsonVal]:
        yield from yaml.safe_load_all(file)
```

Reply:
Different styles, I guess. There's no risk in the current implementation, so fine by me.